### PR TITLE
shop work added

### PR DESCRIPTION
**Pull request by diode02**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (diode02) [inline .gitignore]: add empty line

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  <title>Express</title>
+  <link rel="stylesheet" href="/stylesheets/style.css">
+</head>
+
+<body>
+  <h1>Express</h1>
+  <p>Welcome to Express</p>
+</body>
+
+</html>
```
Comment (diode02) [inline public/index.html]: remove this

File: public/stylesheets/style.css [hunk 3]
```diff
@@ -0,0 +1,8 @@
+body {
+  padding: 50px;
+  font: 14px "Lucida Grande", Helvetica, Arial, sans-serif;
+}
+
+a {
+  color: #00B7FF;
+}
```
Comment (diode02) [inline public/stylesheets/style.css]: remove

File: src/db/mongoose.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const mongoose = require("mongoose");
+const validator = require("validator");
```
Comment (diode02) [inline src/db/mongoose.js]: remove

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+var router = express.Router();
+const { json } = require("express");
+require("./db/mongoose");
+const userRouter = require("./routes/user.route");
+const shopRouter = require("./routes/shop.route");
+const indexRouter = require("./routes/index");
+const addressRouter = require("./routes/address.route");
+const productRouter = require("./routes/product.route");
+const jwt = require("jsonwebtoken");
+const app = express();
```
Comment (diode02) [inline src/index.js]: fix style

File: src/models/address.model.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    },
+  },
+  {
+    timestamps: true,
+  }
+);
+addressSchema.pre("save", async function (next) {
+  const shop = this;
+  next();
+});
```
Comment (diode02) [inline src/models/address.model.js]: remove this

File: src/models/address.model.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+const mongoose = require("mongoose");
+
+const addressSchema = mongoose.Schema(
+  {
+    shopNo: {
+      type: String,
+      required: true,
+    },
+    plaza: {
+      type: String,
```
Comment (diode02) [inline src/models/address.model.js]: add floor

File: src/models/address.model.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    },
+    city: {
+      type: String,
+      required: true,
+    },
+    state: {
+      type: String,
+      required: true,
+    },
+    zipcode: {
```
Comment (diode02) [inline src/models/address.model.js]: add some restriction

File: src/models/address.model.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      type: String,
+      required: true,
```
Comment (diode02) [inline src/models/address.model.js:30]: add validations

File: src/models/address.model.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
+const mongoose = require("mongoose");
+
+const addressSchema = mongoose.Schema(
+  {
+    shopNo: {
+      type: String,
+      required: true,
```
Comment (diode02) [inline src/models/address.model.js:7]: add validation for spaces

File: src/models/shop.model.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+      unique: true,
+    },
+    open: {
+      type: Boolean,
+    },
+    address: {
+      type: [mongoose.Schema.Types.ObjectId],
+      ref: "Address",
+      required: true,
+    },
```
Comment (diode02) [inline src/models/shop.model.js]: remove array

File: src/models/product.model.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+      required: true,
+    },
+    price: {
+      type: Number,
+      required: true,
+    },
+    status: {
+      type: String,
+      required: true,
+      enum: ["sold out", "availible"],
```
Comment (diode02) [inline src/models/product.model.js]: spell correction

**Review by mhaseeb12531 — COMMENTED** [review]
done

**Comment by diode02** [comment]
Will review first thing in morning. Make sure nothing is left from your side

File: .gitignore (hunk 1 above)
Comment (diode02) [reply .gitignore]: add empty line

File: src/index.js [hunk 13]
```diff
@@ -0,0 +1,26 @@
+var express = require("express");
+const { json } = require("express");
+require("./db/mongoose");
+const jwt = require("jsonwebtoken");
+
+const userRouter = require("./routes/user.route");
+const shopRouter = require("./routes/shop.route");
+const indexRouter = require("./routes/index");
+const addressRouter = require("./routes/address.route");
```
Comment (diode02) [inline src/index.js]: remove unused

File: src/index.js [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+const productRouter = require("./routes/product.route");
+
+const app = express();
+var router = express.Router();
+
+app.use(express.json());
```
Comment (diode02) [inline src/index.js:13]: ?

File: src/middleware/auth.js [hunk 15]
```diff
@@ -0,0 +1,87 @@
+const jwt = require("jsonwebtoken");
+const User = require("../models/user.model");
+
+const authAdmin = async (req, res, next) => {
+  try {
+    const token = req.header("Authorization").replace("Bearer ", "");
+    const decoded = jwt.verify(token, "mysignatured");
```
Comment (diode02) [inline src/middleware/auth.js:7]: use env

File: src/models/product.model.js [hunk 16]
```diff
@@ -0,0 +1,34 @@
+const mongoose = require("mongoose");
+
+const userSchema = mongoose.Schema(
```
Comment (diode02) [inline src/models/product.model.js]: typo fix

File: src/routes/shop.route.js [hunk 17]
```diff
@@ -77,7 +90,7 @@ router.patch("/shop-product/:id", authUser, async (req, res) => {
   try {
     const shop = await Shop.findByIdAndUpdate(
       req.params.id,
-      { $push: { product: "625a309d6c06a2c9e53b56c6" } },
+      { $push: { product: "625cc31edc532b9bf8464827" } },
```
Comment (diode02) [inline src/routes/shop.route.js]: ?????

**Pull request merged**